### PR TITLE
Use Go 1.19 fmt for Gitea 1.17, sync emoji data

### PR DESCRIPTION
**Pull request by wxiaoguang**
Recently, the images used by Gitea's drone pipeline were upgraded to Go 1.19.x

It causes the lint fails because Go 1.19 uses new code format.

This PR partially backport #20758 (including the emoji-data sync), partially fix the format manually.

### CODE REVIEW

**Review by KN4CK3R — APPROVED** [review]

File: build/generate-emoji.go [hunk 1]
```diff
@@ -214,8 +214,7 @@ const hdr = `
```
Comment (delvh) [inline build/generate-emoji.go:214]: The `2020` three lines above is now outdated

Comment (delvh) [reply build/generate-emoji.go:214]: But we should probably fix this on main rather then on this branch.

**Review by delvh — APPROVED** [review]

File: build/generate-emoji.go (hunk 1 above)
Comment (wxiaoguang) [reply build/generate-emoji.go:214]: Yup, this is only a backport

**Comment by wxiaoguang** [comment]
The emoji-data is also synchronized from main branch, I think it's fine to get it in 1.17.

If there is some worries, we can revert it later.

**Pull request merged**